File: Python Leet/350.py
import math
from typing import List
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res =[]
        d ={}
        for i in nums1:
            d[i] = d.get(i,0)+1
        
        for i in nums2:
            if i in d:
                d[i] -=1
                if d[i]==0:
                    del d[i]
                res.append(i)
        
        return res

File: leetcode/350.cpp
// leetcode 350 - intersection of two arrays 2 - use a simple map approach


#include<bits/stdc++.h>
#include<algorithm>
#include<string>

using namespace std;

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int,int> mp;
        vector<int> result;
        for(auto i : nums1){
            mp[i]++;
        }
        
        for(auto i : nums2){
            if(mp.count(i)){
                mp[i]--;
                if(mp[i]==0){
                    mp.erase(i);
                }
                result.push_back(i);
            }
        }
        
        return result;
    }
};
